### PR TITLE
TaskBoard component styles

### PR DESCRIPTION
**Pull request by Juveniel**
targets: #2227
design: https://github.com/telerik/kendo-ux-private/issues/281

### CODE REVIEW

File: packages/default/scss/taskboard/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        left: 0;
+    }
+
+    // Cards
+    .k-taskboard-card {
+        @include border-radius( $taskboard-card-border-radius );
+        border-width: $taskboard-card-border-width;
+
+        &.k-taskboard-card-category {
+            border-left-width: $taskboard-card-category-border-width;
```
Comment (PreslavKozovski) [inline packages/default/scss/taskboard/_layout.scss:158]: Should we change the width to border-right in RTL?

File: packages/default/scss/taskboard/_theme.scss [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    // Columns
+    .k-taskboard-column {
+        @include fill(
+            $taskboard-column-text,
+            $taskboard-column-bg,
+            $taskboard-column-border
+        );
+    }
+    .k-taskboard-column.k-state-focus,
+    .k-taskboard-column:focus  {
```
Comment (PreslavKozovski) [inline packages/default/scss/taskboard/_theme.scss:32]: This is not a problem, but let's be consistent :)
```
    .k-taskboard-column:focus,
    .k-taskboard-column.k-state-focus {
```

File: tests/visual/taskboard-card.html [hunk 3]
```diff
@@ -0,0 +1,217 @@
+<!doctype html>
+<html lang="en" class="k-typography">
+<head>
+    <title>TaskBoard</title>
```
Comment (PreslavKozovski) [inline tests/visual/taskboard-card.html:4]:
```suggestion
    <title>TaskBoard Card</title>
```

File: tests/visual/taskboard-card.html [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    <meta charset="utf-8" />
+    <link rel="stylesheet" data-role="kendo-theme" href="../../packages/default/dist/all.css" />
+    <link rel="stylesheet" href="assets/styles.css" />
+    <script src="assets/scripts.js"></script>
+
+    <style>
+        #test-area {}
+    </style>
```
Comment (PreslavKozovski) [inline tests/visual/taskboard-card.html:12]:
```suggestion
```

File: tests/visual/taskboard-card.html [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+</head>
+<body>
+    <div id="test-area" class="grid k-grid-cols-1">
```
Comment (PreslavKozovski) [inline tests/visual/taskboard-card.html:15]: This is not relevant for this specific test :) 

If we use Kendo utils for our tests, why do we continue using `grid` and not - `k-d-grid`? @joneff thought?

File: tests/visual/taskboard-column.html [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+    <meta charset="utf-8" />
+    <link rel="stylesheet" data-role="kendo-theme" href="../../packages/default/dist/all.css" />
+    <link rel="stylesheet" href="assets/styles.css" />
+    <script src="assets/scripts.js"></script>
+
+    <style>
+        #test-area {
+
+        }
+    </style>
```
Comment (PreslavKozovski) [inline tests/visual/taskboard-column.html:14]:
```suggestion
```

File: tests/visual/taskboard-pane.html [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+<head>
+    <title>TaskBoard Preview/Edit Pane</title>
+    <meta charset="utf-8" />
+    <link rel="stylesheet" data-role="kendo-theme" href="../../packages/default/dist/all.css" />
+    <link rel="stylesheet" href="assets/styles.css" />
+    <script src="assets/scripts.js"></script>
+
+    <style>
+        #test-area {}
+    </style>
```
Comment (PreslavKozovski) [inline tests/visual/taskboard-pane.html:12]:
```suggestion
```

File: tests/visual/taskboard-resources.html [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+<head>
+    <title>TaskBoard Resources</title>
+    <meta charset="utf-8" />
+    <link rel="stylesheet" data-role="kendo-theme" href="../../packages/default/dist/all.css" />
+    <link rel="stylesheet" href="assets/styles.css" />
+    <script src="assets/scripts.js"></script>
+
+    <style>
+        #test-area {}
+    </style>
```
Comment (PreslavKozovski) [inline tests/visual/taskboard-resources.html:12]:
```suggestion
```

File: tests/visual/taskboard-rtl.html [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+<head>
+    <title>TaskBoard RTL</title>
+    <meta charset="utf-8" />
+    <link rel="stylesheet" data-role="kendo-theme" href="../../packages/default/dist/all.css" />
+    <link rel="stylesheet" href="assets/styles.css" />
+    <script src="assets/scripts.js"></script>
+
+    <style>
+        #test-area { padding: 0; }
+    </style>
```
Comment (PreslavKozovski) [inline tests/visual/taskboard-rtl.html:12]:
```suggestion
    <style>
        #test-area { padding: 0; }
        
        #test-area > span { padding: 20px 20px 0; }
    </style>
```

File: tests/visual/taskboard.html [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+<head>
+    <title>TaskBoard</title>
+    <meta charset="utf-8" />
+    <link rel="stylesheet" data-role="kendo-theme" href="../../packages/default/dist/all.css" />
+    <link rel="stylesheet" href="assets/styles.css" />
+    <script src="assets/scripts.js"></script>
+
+    <style>
+        #test-area { padding: 0; }
+    </style>
```
Comment (PreslavKozovski) [inline tests/visual/taskboard.html:12]:
```suggestion
    <style>
        #test-area { padding: 0; }
        
        #test-area > span { padding: 20px 20px 0; }
    </style>
```

File: packages/bootstrap/scss/taskboard/_index.scss [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+@import "../mixins/_index.scss";
+
+
+// Dependencies
+@import "../common/_index.scss";
+@import "../icons/_index.scss";
+@import "../forms/_index.scss";
+@import "../input/_index.scss";
+@import "../toolbar/_index.scss";
+@import "../card/_index.scss";
```
Comment (PreslavKozovski) [inline packages/bootstrap/scss/taskboard/_index.scss:11]:
```suggestion
// Dependencies
@import "../utils/_index.scss";
@import "../common/_index.scss";
@import "../icons/_index.scss";
@import "../forms/_index.scss";
@import "../input/_index.scss";
@import "../textbox/_index.scss";
@import "../searchbox/_index.scss";
@import "../toolbar/_index.scss";
@import "../card/_index.scss";
```

**Review by PreslavKozovski — CHANGES_REQUESTED** [review]

File: packages/bootstrap/scss/taskboard/_variables.scss [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+$taskboard-content-padding-y: $taskboard-spacer !default;
+$taskboard-content-padding-x: $taskboard-content-padding-y !default;
+
+$taskboard-column-container-spacing-y: $taskboard-spacer * .75 !default;
+$taskboard-column-container-padding-y: 0 !default;
+$taskboard-column-container-padding-x: $taskboard-spacer * .75 !default;
+$taskboard-columns-container-gap: $taskboard-spacer !default;
+
+$taskboard-column-width: 317px !default;
```
[the remark this answers is not in the document]
Comment (silviyaboteva) [reply packages/bootstrap/scss/taskboard/_variables.scss:27]: In the designs, the width is different in the different themes. I suggest going with a rounded number, for example, 320px and use the same width for the pane.

File: packages/default/scss/taskboard/_layout.scss [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+        flex-shrink: 0;
+        align-self: flex-start;
+        gap: $taskboard-column-header-actions-gap;
+    }
+
+    .k-taskboard-column-cards-container {
+        padding: $taskboard-column-container-padding-y $taskboard-column-container-padding-x;
+        outline: none;
+        overflow: auto;
+        margin-bottom: $taskboard-column-container-spacing-y;
```
Comment (silviyaboteva) [inline packages/default/scss/taskboard/_layout.scss:74]: margin needs to go above the padding

**Review by silviyaboteva — CHANGES_REQUESTED** [review]
Great job @Juveniel ❤️  !
I have only a few comments and two geeneral related to the designs:
 - The card headers should not have a bottom border in Default and Material themes
 - On taskboard-pane test the Edit button should be primary

File: packages/bootstrap/scss/taskboard/_variables.scss (hunk 12 above)
[the remark this answers is not in the document]
Comment (Juveniel) [reply packages/bootstrap/scss/taskboard/_variables.scss]: I changed the width to 320px and updated the tests

File: tests/visual/taskboard-card.html (hunk 5 above)
Comment (Juveniel) [reply tests/visual/taskboard-card.html]: Changed all tests to utilize the `.k-d-grid` class

**Comment by Juveniel** [comment]
@silviyaboteva 

* the button in the pane tests is now updated to primary

* I think the issue with the card border is related to the design(the component uses theme-related card layout). I will sync with the design team just in case and update the border accordingly with a separate PR (if needed).

**Review by PreslavKozovski — APPROVED** [review]

**Review by silviyaboteva — APPROVED** [review]

**Review by joneff — APPROVED** [review]

**Pull request merged**